test/test_hs015.py

```diff
@@ -129,6 +129,30 @@ def test_optimization(self, optName):
         # Check informs
         self.assert_inform_equal(sol)
 
+    def test_ipopt(self):
+        self.optName = "IPOPT"
+        self.setup_optProb()
+        optOptions = self.optOptions.pop(self.optName, None)
+        sol = self.optimize(optOptions=optOptions, storeHistory=True)
+        # Check Solution
+        self.assert_solution_allclose(sol, self.tol[self.optName])
+        # Check informs
+        self.assert_inform_equal(sol)
+
+        # Check iteration counters
+        hist = History(self.histFileName, flag="r")
+        data_init = hist.getValues(names=["iter"], callCounters=[0], allowSens=True)
```

Comment:
Could you add a similar test via the `read` function? I want to make sure that the history file is "correct" and the `read` function should be used instead of `getValues` since it does a bunch of other things.

Reply:
Replaced `getValues` with `read` for the `iter`checks.
I didn't replace it for the second part (where I loop over the iteration and check the consecutive objective values) because we do want to test `getValues` there.

Reply:
Yep looks good to me